### PR TITLE
fix: upgrade liger to 0.5.8 and use native Gemma3 patches

### PR DESCRIPTION
**Pull request by chiwanpark**
# Description

Upgrade Liger-Kernel 0.5.8 ([0.5.7 release notes](https://github.com/linkedin/Liger-Kernel/releases/tag/v0.5.7), [0.5.8 release notes](https://github.com/linkedin/Liger-Kernel/releases/tag/v0.5.7)), and use native Gemma 3 monkey patches instead of custom patches.

## How has this been tested?

I tested my PR by running SFT on Gemma 3 27B.

### CODE REVIEW

**Review by winglian — APPROVED** [review]
thanks! should be good to go once CI is green ✅

**Review by NanoCode012 — APPROVED** [review]

**Comment by NanoCode012** [comment]
Hey @chiwanpark , nice pr. Do you perhaps have the vram usage before/after with this patch?

**Comment by chiwanpark** [comment]
@NanoCode012 I'll measure the VRAM usage difference and post it in tomorrow.

**Comment by chiwanpark** [comment]
@NanoCode012 Sorry for the late posting. I've compared the VRAM usage before/after applying this patch. I tried to train Gemma 3 4B with 16K context sized training set on a A100 80GB card. The effective batch size is 2 (gradient accumulation step: 2, batch size: 1).

Before (liger_fused_linear_cross_entropy: false, liger_cross_entropy: true): 80190MB
After (liger_fused_linear_cross_entropy: true, liger_cross_entropy: false): 67036MB
